lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -4916,6 +4916,17 @@ bool LLVMToSPIRVBase::transExecutionMode() {
       case spv::ExecutionModeNumSIMDWorkitemsINTEL:
       case spv::ExecutionModeSchedulerTargetFmaxMhzINTEL:
       case spv::ExecutionModeMaxWorkDimINTEL:
+      case spv::ExecutionModeRegisterMapInterfaceINTEL: {
+        if (!BM->isAllowedToUseExtension(
+                ExtensionID::SPV_INTEL_kernel_attributes))
+          break;
+        AddSingleArgExecutionMode(static_cast<ExecutionMode>(EMode));
+        BM->addExtension(ExtensionID::SPV_INTEL_kernel_attributes);
+        // CapabilityFPGAKernelAttributesv2INTEL implicitly defines
+        // CapabilityFPGAKernelAttributesINTEL
+        BM->addCapability(CapabilityFPGAKernelAttributesINTEL);
+        BM->addCapability(CapabilityFPGAKernelAttributesv2INTEL);
```

Comment:
The code like this or by splitting switch case:
```suggestion
        if (EMode == spv::ExecutionModeRegisterMapInterfaceINTEL)
          BM->addCapability(CapabilityFPGAKernelAttributesv2INTEL);
```
(note, the suggestion might be no clang-format friendly).

Reply:
Wait isn't this switch case already a switch on 'EMode'? So if control is here then EMode is definitely RegisterMapInterfaceINTEL so isn't this if condition trivially true always?

Reply:
Like this `CapabilityFPGAKernelAttributesv2INTEL` will be emitted for for example `ExecutionModeMaxWorkDimINTEL` as well. If I read the spec correctly, it shouldn't be the case.

Reply:
You are right, I missed the fall-through in the cases above. Thanks for catching this!